client_integration_test.go

```diff
@@ -1249,55 +1249,50 @@ func TestClientReadSequential(t *testing.T) {
 	}
 }
 
+// this writer requires maxPacket = 3 and always returns an error for the second write call
 type lastChunkErrSequentialWriter struct {
-	expected      int
-	written       int
-	writtenReturn int
+	writeCounter int
 }
 
 func (w *lastChunkErrSequentialWriter) Write(b []byte) (int, error) {
-	chunkSize := len(b)
-	w.written += chunkSize
-	if w.written == w.expected {
-		return w.writtenReturn, errors.New("test error")
+	if len(b) > 3 {
+		return 0, errors.New("this writer require maxPacket <= 3, please set MaxPacketChecked(2)")
```

Comment:
The error message here is unclear. The first time this runs, if `maxPacket > 3` then we haven’t set `MaxPacketChecked(3)`, and we _need_ this to be `MaxPacketChecked(3)` not just any other value. Maybe, we add the check in `w.writeCounter == 1` block to ensure that `len(b) == 3` for the first call? That should be a strong enough assertion to avoid all other values than 3.